### PR TITLE
Correct file structure to use button element

### PR DESCRIPTION
**Pull request by lukewarlow**
Currently it's using divs but it should use the actual elements like checkbox does for its label.

### CODE REVIEW

File: site/src/pages/components/file.mdx [hunk 1]
```diff
@@ -95,12 +95,12 @@ button invokes the file selection prompt.
 
 ```html
 <host>
-  <div part="file-selector-button">
+  <button part="file-selector-button">
     <slot name="file-selector-button"></slot>
-  </div>
-  <div part="label">
+  </button>
+  <label part="label">
```
Comment (scottaohara) [inline site/src/pages/components/file.mdx:101]: this seems odd.  is this really its 'label', or is it more a description of the current state of the control.  e.g., no file chosen, x file(s) chosen?  If that latter, this probably shouldn't be a label element then.

Comment (lukewarlow) [reply site/src/pages/components/file.mdx:101]: As per https://gist.github.com/lukewarlow/2eeb75ba6d3039cd10d850c19162976b It's a `<label>` in Firefox. A `<span>` in Blink and not represented in the Dom structure in WebKit. 

Having said that I'm happy to change it to `<span>` (or leave it be for now) if that's considered more appropriate.

Comment (scottaohara) [reply site/src/pages/components/file.mdx:101]: yeh, i don't really see the value in what firefox has there.  I'm actually rather [befuddled by chrome's implementation](https://github.com/w3c/html-aam/issues/421) - this actually reminded me i have bugs to file

Comment (lukewarlow) [reply site/src/pages/components/file.mdx]: I've changed it to be a span to match the DOM structure of Chrome (and it's an inline element so it matches the actual rendering unlike a div)

**Comment by lukewarlow** [comment]
It's worth noting that the slots section actually explicitly calls this out as a label element. @scottaohara should I change that to refer to a span?

image

**Comment by scottaohara** [comment]
yeh, for the fallback content.  maybe the slot name should be updated to help clarify this is more of a prompt/description.  a label serves a different purpose, and this elemement _should_ be labelable as well.

**Comment by lukewarlow** [comment]
Closing in favour of https://github.com/openui/open-ui/pull/913

**Pull request closed without merging**